tfjs-tfdf/src/tfdf_web_api_client.js

```diff
@@ -22,4 +22,17 @@
 // imported correctly by typescript code.
 const YggdrasilDecisionForests = require('../wasm/inference.js');
 
-exports.tfdfWeb = YggdrasilDecisionForests();
+let tfdfWeb = null;
+
+// Function is used here instead of variable to make loading lazy in case user
+// uses setLocateFile before module is used.
+exports.tfdfWeb = () => {
+  if (tfdfWeb == null) {
+    tfdfWeb = YggdrasilDecisionForests();
+  }
+  return tfdfWeb;
+};
+
+exports.setLocateFile = (locateFile) => {
+  tfdfWeb = YggdrasilDecisionForests({locateFile});
+};
```

Comment:
Should we throw an error if the user calls this twice?

Reply:
The user can't create tfdfWeb, as for calling setLocateFile twice it would be similar behavior to tflite's setWasmPath which reassigns until the path is used. I do agree with storing the function though so that later accidental sets don't recreate needlessly, once library is loaded without error we are done.